src/main/java/com/ibm/crypto/plus/provider/RSASignature.java

```diff
@@ -47,6 +49,23 @@ protected void engineInitVerify(PublicKey publicKey) throws InvalidKeyException
             throw new InvalidKeyException("Key is not an RSAPublicKey");
         }
 
+        try {
+            if (provider.isFIPS()) {
+                RSAKeyFactory.checkKeyLengths(((java.security.interfaces.RSAPublicKey) publicKey).getModulus().bitLength(),
```

Comment:
Do we need to add a few tests that try some key values with the FIPS provider that must be in the list `RSAKeyFactory.ALLOWABLE_MODLEN_FIPS_VERIFY)` or else we get an exception. I didnt notice any new tests for this area along with the new `checkKeyLengths` implemented above.

Reply:
Good point there, should test specific key values. Adding tests

Reply:
tests added